### PR TITLE
WIP Splunk module

### PR DESCRIPTION
**Pull request by kelson-martins**
Initial WIP implementation of Splunk module.

### CODE REVIEW

**Comment by delliott90** [comment]
Just a heads up that we want to wait until Mitre gets their stix-translation module merged into six-shifter before merging this PR. There may need to be some changes to this PR once that merge happens since we will no longer be using the external translation module.

**Comment by HedleyWard** [comment]
Updating PR after rebase of merge

**Comment by HedleyWard** [comment]
Updating PR - I have downgraded stix2-validator to version 0.5.0 due to the following issue with version 1.0.1 https://github.com/oasis-open/cti-stix-validator/issues/66

File: stix_shifter/src/modules/splunk/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Splunk
+
+### Format for calling stix-shifter from the command line
+
+python stix_shifter.py `<translator_module>` `<query or result>` `<stix identity object>` `<data>`
+
+(Note the identity object is only used when converting from AQL to STIX, but due to positional arguments, an empty hash will need to be passed in when converting from STIX patterns to AQL. Keyword arguments should be implemented to overcome this).
```
Comment (delliott90) [inline stix_shifter/src/modules/splunk/README.md]: This should probably be changed to read SPLUNK to STIX. I realize this was coming from a copy and paste. Ideally we could rework things so we don't need to pass in that empty hash, and so not even need the note.

Comment (HedleyWard) [reply stix_shifter/src/modules/splunk/README.md]: I've updated the PR and added content to the Readme in the Splunk Module.

**Pull request merged**